sdk/modelsrepository/azure-iot-modelsrepository/src/test/java/com/azure/iot/modelsrepository/DtmiConventionTests.java

```diff
@@ -53,7 +49,24 @@ public void getModelUriTests(String repository, String expectedUri) throws URISy
         URI modelUri = DtmiConventions.getModelUri(dtmi, repositoryUri, false);
         Assertions.assertEquals(expectedUri, modelUri.toString());
     }
-    
+
+    private static Stream<Arguments> getModelUriTestsSupplier() {
+        return Stream.of(
+            Arguments.of("https://localhost/repository/",
+                "https://localhost/repository/dtmi/com/example/thermostat-1.json"),
+            Arguments.of("https://localhost/REPOSITORY",
+                "https://localhost/REPOSITORY/dtmi/com/example/thermostat-1.json"),
+            Arguments.of("file:///path/to/repository/",
+                "file:///path/to/repository/dtmi/com/example/thermostat-1.json"),
+            Arguments.of("file://path/to/RepoSitory", "file://path/to/RepoSitory/dtmi/com/example/thermostat-1.json")
+
+            // TODO: These were disabled as they fail in Linux and macOS, likely due to '\' in the URI.
+//            Arguments.of("C:\\path\\to\\repository\\",
+//                "file:///C:/path/to/repository/dtmi/com/example/thermostat-1.json"),
+//            Arguments.of("\\\\server\\repository", "file:////server/repository/dtmi/com/example/thermostat-1.json")
```

Comment:
@azabbasi these we disabled as they are failing in Linux and macOS, likely due to \ being in the URI

Reply:
I am working on fixing these in the PR I have open
you can merge this in if they are causing issues and cannot wait for the fix.